### PR TITLE
Pagination 3

### PR DESCRIPTION
**Pull request by ITSurgeon**
deleted unusable findAll and comments

### CODE REVIEW

File: src/user/user.service.ts [hunk 1]
```diff
@@ -10,12 +11,31 @@ export class UserService {
...
-    if (users) {
-      return users;
+  async search(query: PaginationQuery) {
+    const newQuery = { page: 1, limit: 9, ...query };
+    const { search, limit, page, ...where } = newQuery;
+    const skip = (page - 1) * limit;
+
+    const builder = this.userRepository
+      .createQueryBuilder('user')
+      // .leftJoinAndSelect('user.categories', 'categories')
```
Comment (evgeniidatsiuk) [inline src/user/user.service.ts:21]: remove it or use it

File: tsconfig.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "baseUrl": "./",
     "incremental": true,
     "skipLibCheck": true,
-    "noImplicitAny": true,
+//    "noImplicitAny": true,
```
Comment (evgeniidatsiuk) [inline tsconfig.json:15]: remove it or use it

File: src/service/service.service.ts [hunk 3]
```diff
@@ -29,12 +35,32 @@ export class ServiceService {
...
+
+    builder.offset(skip);
+    builder.limit(limit);
+
+    const services: Service[] = await builder.getMany();
+
+    return { totalCount, services };
   }
 
+  ///////////////////////////////
```
Comment (evgeniidatsiuk) [inline src/service/service.service.ts:63]: remove

File: src/service/service.controller.ts [hunk 4]
```diff
@@ -21,22 +24,33 @@ export class ServiceController {
...
+        services,
+        totalCount,
+        currentPage: query.page || 1,
+      },
+    };
   }
 
   @Get(':id')
-  findOne(@Param('id') id: string) {
+  async findOne(@Param('id') id: string): Promise<Service> {
```
Comment (evgeniidatsiuk) [inline src/service/service.controller.ts:40]: can be without async

File: src/service/service.controller.ts [hunk 5]
```diff
@@ -21,22 +24,33 @@ export class ServiceController {
   }
 
   @Get()
-  findAll() {
-    return this.serviceService.findAll();
+  async findAll(@Query() query: PaginationQuery) {
```
Comment (evgeniidatsiuk) [inline src/service/service.controller.ts:27]: add type for result

File: src/category/category.service.ts [hunk 6]
```diff
@@ -29,10 +36,29 @@ export class CategoryService {
     return await this.categoryRepository.save(definition);
   }
 
-  async findAll(): Promise<Category[]> {
-    return await this.categoryRepository.find({
-      relations: ['services', 'users'],
-    });
+  async search(query: PaginationQuery) {
```
Comment (evgeniidatsiuk) [inline src/category/category.service.ts:39]: add type for result

File: src/category/category.controller.ts [hunk 7]
```diff
@@ -21,8 +23,16 @@ export class CategoryController {
   }
 
   @Get()
-  findAll() {
-    return this.categoryService.findAll();
+  async findAll(@Query() query: PaginationQuery) {
```
Comment (evgeniidatsiuk) [inline src/category/category.controller.ts:26]: add type for result

File: src/category/category.service.ts [hunk 8]
```diff
@@ -29,10 +36,29 @@ export class CategoryService {
...
+    const newQuery = { page: 1, limit: 9, ...query };
+    const { search, limit, page, ...where } = newQuery;
+    const skip = (page - 1) * limit;
+
+    const builder: SelectQueryBuilder<Category> = this.categoryRepository
+      .createQueryBuilder()
+      .where(where);
```
Comment (evgeniidatsiuk) [inline src/category/category.service.ts:46]: what will be in where?

**Review by evgeniidatsiuk — CHANGES_REQUESTED** [review]

File: src/category/category.service.ts (hunk 8 above)
Comment (ITSurgeon) [reply src/category/category.service.ts:46]: все, що в query, крім page, limit та search

Comment (evgeniidatsiuk) [reply src/category/category.service.ts:46]: @ITSurgeon for example?)

**Review by evgeniidatsiuk — CHANGES_REQUESTED** [review]

File: src/category/category.service.ts (hunk 8 above)
Comment (ITSurgeon) [reply src/category/category.service.ts:46]: For example: name: "ophthalmology"

Comment (evgeniidatsiuk) [reply src/category/category.service.ts:46]: PaginationQuery doesn't exist name field.

**Review by evgeniidatsiuk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**